tests/core/recipes/vasp_recipes/mocked/test_vasp_recipes.py

```diff
@@ -19,6 +24,91 @@
 
 DEFAULT_SETTINGS = SETTINGS.model_copy()
 
+def test_nscf_job(tmp_path, monkeypatch, caplog):
+    monkeypatch.chdir(tmp_path)
+
+    atoms = bulk("Al")
+    calc_kwargs = {}
+    kpoints_mode = "uniform"
+    calculate_optics=True
+
+    output = nscf_job(
+        atoms,
+        prev_dir=tmp_path,
+        bandgap=0.0,
+        nbands_factor=1.0,
+        preset="BulkSet",
+        kpoints_mode=kpoints_mode,
+        calculate_optics=calculate_optics,
+        **calc_kwargs,
+    )
+
+    # Assertions on the output dictionary
+    assert "nsites" in output
+    assert "parameters" in output
+    assert "results" in output
+
+    assert output["parameters"]["ismear"] == -5
+    if calculate_optics:
+        assert output["parameters"]["loptics"] is True
+        assert output["parameters"]["lreal"] is False
+        assert output["parameters"]["cshift"] == 1e-5
+    assert output["parameters"]["lorbit"] == 11
+    assert output["parameters"]["lwave"] is False
+    assert output["parameters"]["lcharg"] is False
+    assert output["parameters"]["nsw"] == 0
+    uniform_isym = 2 if kpoints_mode == "uniform" else 0
+    assert output["parameters"]["isym"] == uniform_isym
+    assert output["parameters"]["icharg"] == 11
+    assert output["parameters"].get("kspacing") is None
+
+    # test nbands and log when vasprun.xml doesn't exist
+    vasprun_exists = False
+    if vasprun_exists:
+        assert "nbands" in output["parameters"]
+    else:
+        assert output["parameters"].get("nbands") is None
+        assert "vasprun.xml* file does not exist in the specified directory." in caplog.text
```

Comment:
Since you are defining `vasprun_exists` as `False`, we don't need an `if`/`else` block. It is also best practice to avoid including such logic statements in tests to help with debugging.